clang/lib/Frontend/InitPreprocessor.cpp

```diff
@@ -1145,6 +1145,7 @@ static void InitializePredefinedMacros(const TargetInfo &TI,
   // SYCL device compiler which doesn't produce host binary.
   if (LangOpts.SYCLIsDevice) {
     Builder.defineMacro("__SYCL_DEVICE_ONLY__", "1");
+    Builder.defineMacro("__SYCL_GLOBAL_VAR__", "__attribute__((sycl_global_var))");
```

Comment:
When I heard the suggestion to wrap it in a macro, I thought that meant "by the SYCL headers" and not as a predefined macro. I don't think this is something we should predefine for users (though, it's at least better than the awful `SYCL_EXTERNAL` macro that happily steals an unreserved identifier from users).

Reply:
A pre-defined macro is what @s-kanaev asked for. If we agree to let the SYCL headers handle it, I'd prefer that approach as well.

Reply:
That would be my preference -- there doesn't seem to be a strong need for the compiler to provide this as a predefined macro.

Reply:
@s-kanaev Thoughts? Perhaps device lib could do:

    #ifdef __SYCL_DEVICE_ONLY__
    #define __SYCL_GLOBAL_VAR__ __attribute__((sycl_global_var))
    #else
    #define __SYCL_GLOBAL_VAR__
    #endif

Then frontend wouldn't need to pre-define the macro.

Reply:
Sounds good.